external/llvm-project/mlir/lib/Dialect/Async/Transforms/AsyncToAsyncRuntime.cpp

```diff
@@ -237,17 +239,66 @@ static Block *setupSetErrorBlock(CoroMachinery &coro) {
   return coro.setError;
 }
 
+/// Build coroutine logic in an outlined function.
```

Comment:
This is a note to me to poke at this, work out how it differs from the handling for exec, and trying to figure out the bug